src/components/PostDetails/PostDetails.tsx

```diff
@@ -1,45 +1,108 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { NewCommentForm } from '../NewCommentForm';
 import './PostDetails.scss';
+import { Post, Comment } from '../../react-app-env';
+import { getComments, delComment } from '../../api/comments';
+import { getPostbyId } from '../../api/posts';
 
-export const PostDetails: React.FC = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
-
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
-
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
-
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
-
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>sad sds dfsadf asdf asdf</p>
-        </li>
-      </ul>
-    </section>
-
-    <section>
-      <div className="PostDetails__form-wrapper">
-        <NewCommentForm />
-      </div>
-    </section>
-  </div>
-);
+type Props = {
+  postId: number | undefined;
+};
+
+export const PostDetails: React.FC<Props> = ({
+  postId,
+}) => {
+  const [commentsList, setCommentsList] = useState<Comment[] | undefined>([]);
+  const [visiblecomments, setVisiblecomments] = useState(false);
+  const [postDetails, setPostDetails] = useState<Post>();
+
+  const findPost = async () => {
+    if (postId) {
+      const result = await getPostbyId(postId);
+
+      setPostDetails(result);
+    }
+  };
+
+  useEffect(() => {
+    findPost();
+  }, []);
+
+  const findcomments = async () => {
+    if (postId) {
+      const result = await getComments(postId);
+
+      setCommentsList(result);
+    }
+  };
+
+  const deletecomment = async (id: number) => {
+    if (commentsList) {
+      await delComment(id);
+      findcomments();
+    }
+  };
+
+  useEffect(() => {
+    findcomments();
+  }, [postId]);
```

Comment:
you can combine useEffect
```suggestion
  useEffect(() => {
    findcomments();
    findPost();
  }, [postId]);
```